### PR TITLE
Coav-control: Add option to set the UDP port

### PR DESCRIPTION
**Pull request by mbelluzzo**

### CODE REVIEW

File: tools/coav-control/coav-control.cc [hunk 1]
```diff
@@ -52,7 +52,8 @@ int main (int argc, char* argv[])
         cout << "Using Sensor: " << sensor_to_name(opts.sensor) << endl;
     }
 
-    shared_ptr<MavQuadCopter> vehicle = std::make_shared<MavQuadCopter>();
+    shared_ptr<MavQuadCopter> vehicle = opts.port ?
+        std::make_shared<MavQuadCopter>(opts.port) : std::make_shared<MavQuadCopter>();
```
Comment (rchiossi) [inline tools/coav-control/coav-control.cc]: ugly! (but yeah, functional)

File: tools/coav-control/parser.cc [hunk 2]
```diff
@@ -245,6 +248,10 @@ control_options parse_cmdline(int argc, char *argv[])
                 exit(-EINVAL);
             }
 
+        // Port
+        } else if (p.option == "-p" || p.option == "--port") {
+            opts.port = (unsigned int) stoul(p.val, nullptr, 0);
```
Comment (rchiossi) [inline tools/coav-control/parser.cc]: why do you set parameters 2 and 3 here? do we need to support port numbers in anything other than base 10?

**Comment by mbelluzzo** [comment]
Since APM uses TCP, I'm going to remove "UDP" mentions

**Comment by rchiossi** [comment]
MavQuadCopter will always start a UDP connection, it doesn't matter if its APM or PX4.

**Comment by mbelluzzo** [comment]
Merged with proposed change

**Pull request closed without merging**